zproject/default_settings.py

```diff
@@ -45,7 +45,7 @@
 # we leave up to Django's defaults.
 
 # LDAP auth
-AUTH_LDAP_SERVER_URI = ""
+AUTH_LDAP_SERVER_URL = ""
```

Comment:
These are part of the external interface for Zulip, so cannot be freely changed.

https://zulip.readthedocs.io/en/latest/subsystems/settings.html

Probably some, like `EXTERNAL_URL_SCHEME`, which don't appear in documentation, we can just change, but for the others, we will need to decide if we're OK breaking compatibility.

Reply:
Maybe I can add a property function in `zerver/models/realms` to check if the `_URL_` ones exist?

Reply:
Should be possible to do this with a bit of logic in `computed_settings.py`, and then the rest of the codebase can be entirely in terms of the new names.

This looks like one existing example (or pair of examples):
```py
SOCIAL_AUTH_GOOGLE_SECRET = get_secret("social_auth_google_secret")
# Fallback to google-oauth settings in case social auth settings for
# Google are missing; this is for backwards-compatibility with older
# Zulip versions where /etc/zulip/settings.py has not been migrated yet.
GOOGLE_OAUTH2_CLIENT_SECRET = get_secret("google_oauth2_client_secret")
SOCIAL_AUTH_GOOGLE_KEY = SOCIAL_AUTH_GOOGLE_KEY or GOOGLE_OAUTH2_CLIENT_ID
SOCIAL_AUTH_GOOGLE_SECRET = SOCIAL_AUTH_GOOGLE_SECRET or GOOGLE_OAUTH2_CLIENT_SECRET
```

Reply:
Thanks, that is helpful. Also, `URI` appears in the following server settings:
- AUTH_LDAP_SERVER_URI
- DEFAULT_AVATAR_URI
- DEFAULT_LOGO_URI
- CAMO_URI
- EXTERNAL_URI_SCHEME
- REALM_MOBILE_REMAP_URIS

I am not sure which of these need backwards-compatilbilty. Can you please mention them?

Reply:
This PR currently does not replace `URI` in server settings though. Maybe that can be done in a separate PR if that's OK?

Reply:
Yes doing the server settings part in a separate migration is preferred.